### ISSUE TITLE
Package OpenSSL library with iOS builds when required

### ISSUE THREAD

**Issue by peardox** [castle-engine/castle-engine#185]
as per #184 

You do this for things like the font library etc if I recall correctly this being why you need to load the xcworkspace - so both the app and dependencies are built

I remember this from a month+ ago - 2d_dragon_spine??? I'll go look and see if I can work it out myself...

**Comment by peardox** [comment]
Well, I added an OpenSSL Pod to build tool by copying ios/services/freetype to openssl then modiflying the Podfile to...

`pod 'OpenSSL-Universal', :inhibit_warnings => true
`
It built the Pod and it was visible in XCode which compiled and installed OK  but...

App still sulking - stuck on first download

Or am I missing something dumb with the pod deployment?

Is there a similar services CastleManifest for OSX as the pod says it works with everything and is only a few months old - i.e. it should also work for castle_download

**Comment by peardox** [comment]
The now fixed OpenSSL on Mac is no good for iOS 👍 

Examining the fix everything appears to be done via the dylib (version checking was failing on the Mac), no attempt appears to be made to use a static lib (I made one available to the build). There's a LoadUtilLibrary that handles the DLL loading but I can't find a reference to it so presume it's used higher up the api chain.

It's not a situation that can be bypassed by using un-encrypted data either as Apple demand everything takes place over SSL (dumb when it's all public data + images)

**Comment by michaliskambi** [comment]
iOS is a different architecture(s) than desktop macOS, and also it uses static libraries (while on desktop we use dynamic ones). So no surprise that a different solution is necessary. FreeType, OggVorbis loading work like that on iOS now.

Likely iOS needs a different implementation of downloader altogether, to use Objective-C API underneath to download. This would use system's networking (including ssl) libraries. That's what we do on Android, where downloading goes through standard Android Java API, not FpHttpClient.

Does unencrypted http works on iOS (through FpHttpClient) at all?

**Comment by peardox** [comment]
Good point - adequate for testing while we sort out https... Just gave it a go, copied the demo program files to a non-https server I just set up. Odd results - came back with a DNS Resolution error - "Hist name resolution for "nossl.peardox.co.uk" failed for all three tests

Can you check out one or more of the following (no grafix on web pages of course - just did a wget for them all)

http://nossl.peardox.co.uk/latest.zip
http://nossl.peardox.co.uk/modern_pascal_introduction.html
http://nossl.peardox.co.uk/wiki/Main_Page

It's possible that the DNS ain't caught up yet on my iPad + iPhone so if you can get a response that's one thing to rule out.

I've reset the network on both devices

I'll try running the test again later - give it a few hours... 

Looking at a native solution I found this which looks like a starting point - https://developer.apple.com/documentation/foundation/url_loading_system/downloading_files_from_websites

**Comment by peardox** [comment]
Hmm - just tried the links on the Mac - no joy there in either Safari or Chrome

Looks like Apple has DNS dementia

**Comment by michaliskambi** [comment]
I checked the links 

http://nossl.peardox.co.uk/latest.zip
http://nossl.peardox.co.uk/modern_pascal_introduction.html
http://nossl.peardox.co.uk/wiki/Main_Page

from Linux (in Firefox and wget). Then I checked them from macOS Catalina and macOS Mojave systems (using curl and wget). They work fine in all cases (both from Linux and 2 macOS systems).

Also, DNS should be unrelated to whether you later download the resource using http or https. DNS just resolves the domain name to IP, it has to be done regardless if you later use http or https. (And it is independent from whether DNS is done by classic DNS protocol over UDP, or using DNS-over-HTTPS.)

Likely the DNS information wasn't propagated yet, if you created these domain names not much earlier. 

So please try again.

**Comment by michaliskambi** [comment]
> https://developer.apple.com/documentation/foundation/url_loading_system/downloading_files_from_websites

Most probably yes, this is what we should use on iOS, in a new service equivalent to `download_urls` for Android.

**Comment by peardox** [comment]
I can now get the URIs on everything apart from iOS devices via app

iOS via example app with new URIs still gives me the name resolution stuff
Safari on iOS opens everything OK
Recompiling the same code on Win10 got me the expected results
